docs/source/reconciliation.rst

```diff
@@ -0,0 +1,154 @@
+
+    [
+        {"prefix": "a", "uri_prefix": "https://example.org/a/", "prefix_synonyms": ["a1"]},
+        {"prefix": "b", "uri_prefix": "https://example.org/b/"}
+    ]
+
+
+CURIE Prefix Remapping
+----------------------
+CURIE prefix remapping is configured by a mapping from existing CURIE prefixes to new CURIE prefixes.
+The following rules are applied for each pair of old/new prefixes:
+
+1. New prefix exists
+~~~~~~~~~~~~~~~~~~~~
+If the new prefix appears as a prefix synonym in the record corresponding to the old prefix, they are swapped.
+This means applying the CURIE prefix remapping ``{"a": "a1"}`` results in the following
+
+.. code-block:: json
+
+    [
+        {"prefix": "a1", "uri_prefix": "https://example.org/a/", "prefix_synonyms": ["a"]},
+        {"prefix": "b", "uri_prefix": "https://example.org/b/"}
+    ]
+
+If the new prefix appears as a preferred prefix or prefix synonym for any other record, one of two things can happen:
+
+1. Do nothing (lenient)
+2. Raise an exception (strict)
+
+This means applying the CURIE prefix remapping ``{"a": "b"}`` results in either no change or an exception being raised.
+
+2. New prefix doesn't exist, old prefix exists
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+If the old prefix appears in a record in the extended prefix map as a preferred prefix:
+
+1. Replace the record's preferred prefix with the new prefix
+2. Add the record's old preferred prefix to the record's prefix synonyms
+
+This means applying the CURIE prefix remapping ``{"a": "c"}`` results in the following
+
+.. code-block:: json
+
+    [
+        {"prefix": "c", "uri_prefix": "https://example.org/a/", "prefix_synonyms": ["a", "a1"]},
+        {"prefix": "b", "uri_prefix": "https://example.org/b/"}
+    ]
+
+Similarly, if the old prefix appears in a record in the extended prefix map as a prefix synonym, do the same.
+This means applying the CURIE prefix remapping ``{"a1": "c"}`` results in the following
+
+.. code-block:: json
+
+    [
+        {"prefix": "c", "uri_prefix": "https://example.org/a/", "prefix_synonyms": ["a", "a1"]},
+        {"prefix": "b", "uri_prefix": "https://example.org/b/"}
+    ]
+
+3. New prefix doesn't exist, old prefix doesn't exist
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+If neither the old prefix nor new prefix appear in the extended prefix maps, one of two things can happen:
+
+1. Do nothing (lenient)
+2. Raise an exception (strict)
+
+URI Prefix Remapping
+----------------------
+URI prefix remapping is configured by a mapping from existing URI prefixes to new URI prefixes.
+The rules work exactly the same as with CURIE prefix remapping, but for the :data:`curies.Record.uri_prefix` and
+:data:`curies.Record.uri_prefix_synonyms` fields.
+
+Prefix Rewiring
+---------------
+Prefix rewiring is configured by a mapping from existing CURIE prefixes to new URI prefixes.
+The following rules are applied for each pair of CURIE prefix/URI prefix:
+
+1. CURIE prefix doesn't exist
```

Comment:
```suggestion
1. CURIE prefix doesn't exist, URI prefix doesn't exist
```

Reply:
I would distinguish this case clearly from `CURIE prefix doesn't exist, URI prefix exists` because of the risk of hitting a conflict on the URI prefix side.